src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -580,7 +583,8 @@ module.exports = function(
       oldProps === newProps &&
       oldState === newState &&
       !hasContextChanged() &&
-      !(updateQueue !== null && updateQueue.hasForceUpdate)
+      !(workInProgress.updateQueue !== null &&
```

Comment:
`beginUpdateQueue` sometimes clones `workInProgress.updateQueue`, so we need to point to the latest version.